kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/TlsSecrets.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright Kroxylicious Authors.
+ *
+ * Licensed under the Apache Software License version 2.0, available at http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package io.kroxylicious.kubernetes.operator;
+
+import java.nio.file.Path;
+import java.util.List;
+
+import io.kroxylicious.kubernetes.api.v1alpha1.KafkaProxy;
+
+public record TlsSecrets(Path path, String volumeName, String secretName) {
+
+    private static final Path TLS_SECRETS_ROOT_DIR = Path.of("/opt/kroxylicious/secrets/tls");
+    private static final String CERTIFICATE = "tls.crt";
+    private static final String KEY = "tls.key";
+
+    public static List<TlsSecrets> tlsSecretsFor(KafkaProxy proxy) {
+        return proxy.getSpec().getListeners().stream().flatMap(listeners -> listeners.getTls().getCertificateRefs().stream().map(cert -> {
```

Comment:
currently we create one mount per secret and mount in all the key-values from the Secret. It might be better to mount in the two subpaths we are interested in instead, as then the Deployment would fail to come up if they weren't in the secret.

Reply:
IIRC mounting by subpath prevents the mounted secret entries being updated (when the secret contents changes) within the running container. That could be a useful route to (eventually) being able to handle certificate renewals dynamically, without needing to restart the container with the client disconnection and disruption that would imply.

I'm sure with a little care we could arrange the proxy to throw and write the reason to the termination message file, so it was a little more obvious why it was crashlooping.